AFNetworking/AFURLRequestSerialization.m

```diff
@@ -578,8 +578,8 @@ - (void)encodeWithCoder:(NSCoder *)coder {
 
 - (instancetype)copyWithZone:(NSZone *)zone {
     AFHTTPRequestSerializer *serializer = [[[self class] allocWithZone:zone] init];
-    dispatch_sync(self.requestHeaderModificationQueue, ^{
-        serializer.mutableHTTPRequestHeaders = [self.mutableHTTPRequestHeaders mutableCopyWithZone:zone];
+    dispatch_sync(serializer.requestHeaderModificationQueue, ^{
```

Comment:
Add a comment explaining why inside the block we do `self.HTTPRequestHeaders` and not `self.mutableRequestHeaders`. Someone might come in the future and "optimize" this, trying to avoid the double dictionary copy.

Reply:
Done,

Reply:
One question that should be raised here is where do we want to fetch `self.HTTPRequestHeaders` - should we do it here or before the call to `dispatch_sync`? My gut says that we need to move it outside the `dispatch_sync`, since in this case we have two nested synchronous calls. There's of course no possible way for `self` to equal `serializer`, so we will not `dispatch_sync` twice to the same queue, but I learned to be afraid of blocking operations.

Thoughts?

Reply:
We had a discussion exactly about that yesterday and we figured it does not really matter. The nested dispatch_sync is what deterred me too. I tend to extract `self.HTTPRequest` out of the block but this will generate a code that raises more questions (the WTF score of the code increases).
Anyway I asked Neria to add a comment explaining the why we don't use `self.mutableHTTPRequestHeaders` here so we can extract this out of the block.

Reply:
I agree that it doesn't really matter in the context of the fix, but I feel that we should strive to avoid locking of multiple resources at once if possible, therefore I think it should be extracted.